mirrorlist.go

```diff
@@ -10,51 +10,45 @@ import (
 )
 
 func updateMirrorlists() {
-	for name, _ := range config.Repos {
-		err := checkAndUpdateMirrorlistRepo(name)
+	for name, repo := range config.Repos {
+		err := checkAndUpdateMirrorlistRepo(name, repo)
 		if err != nil {
 			log.Fatal(err)
 		}
 	}
 }
 
-func checkAndUpdateMirrorlistRepo(repoName string) error {
-	repo, ok := config.Repos[repoName]
-	if !ok {
-		return fmt.Errorf("repo %v does not exist in config", repoName)
-	}
+func checkAndUpdateMirrorlistRepo(repoName string, repo *Repo) error {
 	if repo.Mirrorlist != "" {
-		lastCheck, ok := lastMirrorlistCheck[repo.Mirrorlist]
-		if ok && time.Since(lastCheck) < 5*time.Second {
+		repo.mutex.Lock()
+		defer repo.mutex.Unlock()
```

Comment:
I'd move the locking after line 29, it is useless IMO to lock it if it won't be modified

Reply:
No, i think we need the lock around the read to prevent `LastMirrorlistCheck` from being written to while it's being read on line 26.

Reply:
You're right, sorry :+1: